File: commons/dto/src/main/java/net/blueshell/dto/GuestDTO.java
package net.blueshell.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.sql.Timestamp;

@Data
@EqualsAndHashCode(callSuper = false)
public class GuestDTO extends BaseDTO {

    private Long id;

    private String name;

    private String discord;

    private String email;

    private Timestamp createdAt;

    private String accessToken;
}

File: api/src/main/kotlin/net/blueshell/api/domain/event/web/dto/GuestDTO.kt
package net.blueshell.api.domain.event.web.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull
import net.blueshell.api.domain.user.web.dto.PersonalInfoDTO

@Schema(name = "Guest")
data class GuestDTO(
    @field:NotNull
    var name: String? = null,
) : PersonalInfoDTO() {
    var accessToken: String? = null
}
